clientv3/integration/dial_test.go

```diff
@@ -66,6 +70,89 @@ func TestDialTLSExpired(t *testing.T) {
 	}
 }
 
+// TestDialTLSExpiredReload ensures server reloads expired certs,
+// rejecting client requests, and vice versa.
+func TestDialTLSExpiredReload(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	ts, err := copyTLSFiles(testTLSInfo)
+	if err != nil {
+		t.Fatal(err)
+	}
+	certsDir := filepath.Dir(ts.KeyFile)
+	defer os.RemoveAll(certsDir)
+
+	tse, err := copyTLSFiles(testTLSInfoExpired)
+	if err != nil {
+		t.Fatal(err)
+	}
+	dir2 := filepath.Dir(tse.KeyFile)
+	defer os.RemoveAll(dir2)
+
+	var tmpDir string
+	tmpDir, err = ioutil.TempDir(os.TempDir(), "fixtures")
+	if err != nil {
+		t.Fatal(err)
+	}
+	os.RemoveAll(tmpDir)
+	defer os.RemoveAll(tmpDir)
+
+	// start with valid certs
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 1, PeerTLS: &ts, ClientTLS: &ts})
+	defer clus.Terminate(t)
+
+	// replace certs directory with expired ones
+	if err = os.Rename(certsDir, tmpDir); err != nil {
+		t.Fatal(err)
+	}
+	if err = os.Rename(dir2, certsDir); err != nil {
+		t.Fatal(err)
+	}
+
+	// 'tmpDir' now has valid certs
+	// 'certsDir' now has expired certs; 'dir2' does not exist
+
+	// now server expects 'tls: bad certificate'
+	// on incoming client requests
+	tls, err := ts.ClientConfig()
```

Comment:
Some concurrency would be good here. Boot a cluster, start a goroutine that spams it with connections,  manipulate the certs while connections concurrently hit the server. The connection goroutine would complete when it transitions from good connections to tls rejects.